homeassistant/components/myq/const.py

```diff
@@ -0,0 +1,32 @@
+"""The MyQ integration."""
+from homeassistant.const import STATE_CLOSED, STATE_CLOSING, STATE_OPEN, STATE_OPENING
+
+DOMAIN = "myq"
+
+PLATFORMS = ["cover"]
+
+MYQ_DEVICE_TYPE = "device_type"
+MYQ_DEVICE_TYPE_GATE = "gate"
+MYQ_DEVICE_STATE = "state"
+MYQ_DEVICE_STATE_ONLINE = "online"
+
+MYQ_TO_HASS = {
+    "closed": STATE_CLOSED,
+    "closing": STATE_CLOSING,
+    "open": STATE_OPEN,
+    "opening": STATE_OPENING,
+}
+
+GATE_STATE_ICONS = {
+    STATE_CLOSED: "mdi:gate",
+    STATE_CLOSING: "mdi:gate-arrow-right",
+    STATE_OPENING: "mdi:gate-arrow-right",
+    STATE_OPEN: "mdi:gate-open",
+}
+
+GARAGE_STATE_ICONS = {
+    STATE_CLOSED: "mdi:garage-variant",
+    STATE_CLOSING: "mdi:arrow-down-bold-box",
+    STATE_OPENING: "mdi:arrow-up-bold-box",
+    STATE_OPEN: "mdi:garage-open-variant",
+}
```

Comment:
This should be covered by the frontend by setting a device class. If it is not, we should fix that in the frontend https://github.com/home-assistant/frontend/blob/dev/src/common/entity/cover_icon.ts

Reply:
Nice.  We probably need DEVICE_CLASS_GATE since we only have DEVICE_CLASS_GARAGE
```
DEVICE_CLASS_AWNING = "awning"
DEVICE_CLASS_BLIND = "blind"
DEVICE_CLASS_CURTAIN = "curtain"
DEVICE_CLASS_DAMPER = "damper"
DEVICE_CLASS_DOOR = "door"
DEVICE_CLASS_GARAGE = "garage"
DEVICE_CLASS_SHADE = "shade"
DEVICE_CLASS_SHUTTER = "shutter"
DEVICE_CLASS_WINDOW = "window"
```

Will work up a new PR for this

Reply:
I hope I did this right:
https://github.com/home-assistant/core/pull/33074
https://github.com/home-assistant/developers.home-assistant/pull/444
https://github.com/home-assistant/frontend/pull/5299